eng/ci/templates/official/jobs/build-test.yml

```diff
@@ -0,0 +1,262 @@
+      TELEMETRY_INSTRUMENTATION_KEY: $(TELEMETRY_INSTRUMENTATION_KEY)
+    condition: and(succeeded(), eq(variables['IsReleaseBuild'], 'true'))
+  - task: DotNetCoreCLI@2
+    inputs:
+      command: 'run'
+      workingDirectory: '.\build'
+      arguments: 'TestSignedArtifacts --signTest'
+    displayName: 'Verify signed binaries'
+    condition: and(succeeded(), eq(variables['IsReleaseBuild'], 'true'))
+  - pwsh: |
+      .\generateMsiFiles.ps1
+    env:
+      AzureBlobSigningConnectionString: $(AzureBlobSigningConnectionString)
+      BuildArtifactsStorage: $(BuildArtifactsStorage)
+      DURABLE_STORAGE_CONNECTION: $(DURABLE_STORAGE_CONNECTION)
+      TELEMETRY_INSTRUMENTATION_KEY: $(TELEMETRY_INSTRUMENTATION_KEY)
+    displayName: 'Generate MSI files'
+    condition: and(succeeded(), eq(variables['IsReleaseBuild'], 'true'))
+  
+  - template: ci/sign-files.yml@eng
+    parameters:
+      displayName: 'MSI Code Signing'
+      folderPath: '$(Build.Repository.LocalPath)\artifacts'
+      pattern: '*.msi'
+      signType: inline
+      inlineOperation: |
+        [
+            {
+              "KeyCode": "CP-230012",
+              "OperationCode": "SigntoolSign",
+              "Parameters": {
+                "OpusName": "Microsoft",
+                "OpusInfo": "http://www.microsoft.com",
+                "FileDigest": "/fd \"SHA256\"",
+                "PageHash": "/NPH",
+                "TimeStamp": "/tr \"http://rfc3161.gtm.corp.microsoft.com/TSS/HttpTspServer\" /td sha256"
+              },
+              "ToolName": "sign",
+              "ToolVersion": "1.0"
+            },
+            {
+              "KeyCode": "CP-230012",
+              "OperationCode": "SigntoolVerify",
+              "Parameters": {},
+              "ToolName": "sign",
+              "ToolVersion": "1.0"
+            }
+        ]
+
+  - pwsh: |
+      .\generateSha.ps1
+    displayName: 'Generate sha files'
+
+  - pwsh: |
+      Move-Item -Path '$(Build.Repository.LocalPath)\artifacts\Azure.Functions.Cli.*' -Destination '$(Build.ArtifactStagingDirectory)'
+      Move-Item -Path '$(Build.Repository.LocalPath)\artifacts\func-cli*.msi' -Destination '$(Build.ArtifactStagingDirectory)'
+      Move-Item -Path '$(Build.Repository.LocalPath)\artifacts\Microsoft.Azure.Functions.CoreTools*.nupkg' -Destination '$(Build.ArtifactStagingDirectory)'
+      if ($env:IntegrationBuildNumber -like "PreRelease*-*")
+      {
+          $sourcePath = Join-Path '$(Build.Repository.LocalPath)\artifacts' 'integrationTestsBuildManifest.json'
+          Move-Item -Path $sourcePath -Destination '$(Build.ArtifactStagingDirectory)'
+      }
+    env:
+      IntegrationBuildNumber: $(INTEGRATIONBUILDNUMBER)
+    displayName: 'Move artifacts'
+  - task: 1ES.PublishPipelineArtifact@1
+    condition: succeeded()
+    inputs:
+      targetPath: '$(Build.ArtifactStagingDirectory)'
+      artifactName: 'drop'
+      artifactType: 'pipeline'
```

Comment:
As far as I know, for publishing artifacts, we are asked to use templateContext. 

Example:https://github.com/Azure/azure-functions-java-worker/blob/387e2eb696f7252bc70e0369f00da4750d1335f8/eng/ci/templates/official/jobs/build-artifacts.yml#L5-L18

Reply:
1ES.PublishPipelineArtifact is also accepted change. I will consider changing in different PR.